File: src/Day02.java
import java.io.IOException;
import java.nio.file.*;
import java.util.*;

public class Day02 {

	public static long partOne(List<String> list) {
		int totalFeet = 0;
		for (String s : list) {
			int[] size = new int[3];
			String[] s_split = s.split("x");
			for (int i = 0; i < s_split.length; i++) 
				size[i] = Integer.valueOf(s_split[i]);
						
			Arrays.sort(size);

			totalFeet += 2 * size[0] * size[1] + 2 * size[0] * size[2] + 2 * size[1] * size[2];
			totalFeet += size[0] * size[1];
		}
		return totalFeet;
	}

	public static long partTwo(List<String> list) {
		int totalFeet = 0;
		for (String s : list) {
			int[] size = new int[3];
			String[] s_split = s.split("x");
			for (int i = 0; i < s_split.length; i++) 
				size[i] = Integer.valueOf(s_split[i]);
			
			Arrays.sort(size);

			totalFeet += 2 * (size[0] + size[1]);
			totalFeet += size[0] * size[1] * size[2];
		}
		return totalFeet;
	}

	public static void main(String[] args) throws IOException {
		List<String> s = Files.readAllLines(Paths.get("./input/Day02_input.txt"));
		System.out.println("Part One = " + partOne(s));
		System.out.println("Part Two = " + partTwo(s));
	}
}

File: src/aoc2017/kot/Day02.kt
package aoc2017.kot

import itertools.times
import to2DIntArr

import java.io.File

object Day02 {
  fun partOne(input: List<List<Int>>): Int = input.map { it.max()!! - it.min()!! }.sum()

  fun partTwo(input: List<List<Int>>): Int {
    return input.map {
      it.times(it).filter { (j, k) -> j != k && j % k == 0 }.map { it.first / it.second }.sum()
    }.sum()
  }
}

fun main(args: Array<String>) {
  val input = File("./input/2017/Day02_input.txt").to2DIntArr("\t")
  println("Part One = ${Day02.partOne(input)}")
  println("Part Two = ${Day02.partTwo(input)}")
}